webpack/webpack-cli
# Next

**What kind of change does this PR introduce?**

refactoring + bugfix + features + other

**Did you add tests for your changes?**

Yes

**If relevant, did you update the documentation?**

Yes

**Summary**

It is the next branch

**Does this PR introduce a breaking change?**

Yes, list of breaking changes will be here in future

**Other information**

TODO:
- ~~drop node v10~~
- ~~drop webpack v4 + aliases for `watch` (i.e. `w`), `devtool`. i.e. `d`, `target` is `t`, `output-path` is `o` were removed~~
- remove `prefetch`
- update all deps to the latest
- `--entry` require `--entry-reset` before (need add test(s))
- ~~`--target` require `--target-reset` before (need add test(s))~~
- refactor `build` tests - avoid using `scripts/prepareSuite.js` and put all deps in top `package.json` + rename tests
- `node-env` should be renamed?
- simplify logic for `getBuiltInOptions` and remove extra `map`/`reduce`/etc and avoid using `builtInOptionsCache`
- make stable CLI API for public usage
- avoid unnecessary `process.exit()`
- resolve TODO where it possible
- remove `webpack version` in favor `webpack info`
